dcos_test_utils/dcos_cli.py

```diff
@@ -92,6 +94,45 @@ def clear_cli_dir():
         if os.path.exists(path):
             shutil.rmtree(path)
 
+    def exec(self, cmd: str, stdin=None, check=True) -> subprocess.CompletedProcess:
+        """Execute CLI command and processes result.
+
+        This method expects that process won't block.
+
+        :param cmd: Program and arguments
+        :type cmd: str
+        :param stdin: File to use for stdin
+        :param check: Does it check for raised errors
+        :type stdin: File
+        :returns: A tuple with stdout and stderr
+        :rtype: subprocess.CompletedProcess
+        """
+
+        log.info('CMD: {!r}'.format(cmd))
+
+        try:
+            process = subprocess.run(
+                cmd,
+                stdin=stdin,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                env=self.env,
+                check=check)
+        except subprocess.CalledProcessError as e:
+            stderr = e.stderr.decode('utf-8')
```

Comment:
Are you sure `utf-8` will work?
Have a look at the DC/OS E2E way of decoding.

https://github.com/dcos/dcos-e2e/blob/8d4916780ade8caf41dae376fdf47f4253eb52c7/src/dcos_e2e/_common.py#L46-L59

Reply:
I think it is a good idea to introduce the _safe_decode. I am also building on the previous behavior as the minimum value for me is returning the process object. I will make this change.